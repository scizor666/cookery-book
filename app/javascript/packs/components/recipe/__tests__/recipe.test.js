import React from 'react';
import {recipe} from './test_data';
import renderer from 'react-test-renderer';
import Recipe from '../recipe';

jest.mock('../recipe_actions', () => 'RecipeActions');
jest.mock('../ingredients_table', () => 'IngredientsTable');

test('display a recipe', () => {
    const wrapper = renderer.create(<Recipe recipe={recipe}/>);
    expect(wrapper.toJSON()).toMatchSnapshot();
});