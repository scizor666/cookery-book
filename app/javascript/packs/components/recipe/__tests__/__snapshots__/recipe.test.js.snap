// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display a recipe 1`] = `
<div
  className="main"
>
  <div
    className="row"
  >
    <div
      className="col col-auto"
    >
      <h1>
        product name here
      </h1>
    </div>
    <RecipeActions
      recipe={
        Object {
          "caloricity": 666,
          "catalog_id": 500,
          "description": "product description here",
          "id": 100,
          "image_url": "http://some.test.url/with_image.png",
          "ingredients": Array [
            Object {
              "id": 1,
              "product": Object {
                "caloricity": 107.7,
                "id": 2,
                "name": "chicken",
              },
              "weight": 300,
            },
            Object {
              "id": 2,
              "product": Object {
                "caloricity": 44,
                "id": 1,
                "name": "carrot",
              },
              "weight": 155,
            },
          ],
          "name": "product name here",
          "short_description": "short product description here",
        }
      }
    />
  </div>
  <div
    className="collapse show"
    id="recipe_details_47"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-12 col-md-6"
      >
        <div
          className="lead"
        >
          short product description here
        </div>
        <img
          alt="No picture found"
          className="w-100"
          src="http://some.test.url/with_image.png"
        />
      </div>
      <IngredientsTable
        caloricity={666}
        ingredients={
          Array [
            Object {
              "id": 1,
              "product": Object {
                "caloricity": 107.7,
                "id": 2,
                "name": "chicken",
              },
              "weight": 300,
            },
            Object {
              "id": 2,
              "product": Object {
                "caloricity": 44,
                "id": 1,
                "name": "carrot",
              },
              "weight": 155,
            },
          ]
        }
      />
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <p
          className="recipe-description lead"
        >
          product description here
        </p>
      </div>
    </div>
  </div>
</div>
`;
